dali/base/dautils.cpp

```diff
@@ -1949,16 +1960,22 @@ class CTimedCache
 
     unsigned check()
     {
+        unsigned expired = 0;
         unsigned res = (unsigned)-1;
         unsigned now = msTick();
-        ForEachItemInRev(i,items) {
+        ForEachItemIn(i, items)
+        {
             CTimedCacheItem &item = items.item(i);
-            unsigned t = item.due-now;
-            if ((int)t<=0)
-                items.remove(i);
-            else if (t<res)
+            unsigned t = item.due - now;
+            if ((int)t > 0)
```

Comment:
minor: not new, but the unsigned / cast is not v. nice.
Would be more readable if:
```
    if (item.due > now)
    {
        res = item.due-now;
        break;
    }
```